### PR TITLE
add resource azurerm_cosmosdb_sql_user_defined_function

### PR DESCRIPTION
**Pull request by ms-henglu**
The added tests are listed as follows.
```
=== RUN   TestAccCosmosDbSQLFunction_basic
=== PAUSE TestAccCosmosDbSQLFunction_basic
=== CONT  TestAccCosmosDbSQLFunction_basic
--- PASS: TestAccCosmosDbSQLFunction_basic (1286.92s)
=== RUN   TestAccCosmosDbSQLFunction_requiresImport
=== PAUSE TestAccCosmosDbSQLFunction_requiresImport
=== CONT  TestAccCosmosDbSQLFunction_requiresImport
--- PASS: TestAccCosmosDbSQLFunction_requiresImport (1300.76s)
=== RUN   TestAccCosmosDbSQLFunction_update
=== PAUSE TestAccCosmosDbSQLFunction_update
=== CONT  TestAccCosmosDbSQLFunction_update
--- PASS: TestAccCosmosDbSQLFunction_update (1342.00s)
PASS
```

### CODE REVIEW

File: azurerm/internal/services/cosmos/cosmosdb_sql_user_defined_function_resource.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"database_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
```
Comment (katbyte) [inline azurerm/internal/services/cosmos/cosmosdb_sql_user_defined_function_resource.go:66]: there is no validation here, also couldn't we just take in a sql_database_id and parse this all out?

File: azurerm/internal/services/cosmos/cosmosdb_sql_user_defined_function_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
```
Comment (katbyte) [inline azurerm/internal/services/cosmos/cosmosdb_sql_user_defined_function_resource_test.go:66]: could we finish with a -> basic step?

File: azurerm/internal/services/cosmos/registration.go [hunk 3]
```diff
@@ -28,16 +28,17 @@ func (r Registration) SupportedDataSources() map[string]*schema.Resource {
...
+		"azurerm_cosmosdb_account":                   resourceCosmosDbAccount(),
+		"azurerm_cosmosdb_cassandra_keyspace":        resourceCosmosDbCassandraKeyspace(),
+		"azurerm_cosmosdb_cassandra_table":           resourceCosmosDbCassandraTable(),
+		"azurerm_cosmosdb_gremlin_database":          resourceCosmosGremlinDatabase(),
+		"azurerm_cosmosdb_gremlin_graph":             resourceCosmosDbGremlinGraph(),
+		"azurerm_cosmosdb_mongo_collection":          resourceCosmosDbMongoCollection(),
+		"azurerm_cosmosdb_mongo_database":            resourceCosmosDbMongoDatabase(),
+		"azurerm_cosmosdb_sql_container":             resourceCosmosDbSQLContainer(),
+		"azurerm_cosmosdb_sql_database":              resourceCosmosDbSQLDatabase(),
+		"azurerm_cosmosdb_sql_user_defined_function": resourceCosmosDbSQLUserDefinedFunction(),
```
Comment (katbyte) [inline azurerm/internal/services/cosmos/registration.go:40]: is user defined really necessary given we are define it?
```suggestion
		"azurerm_cosmosdb_sql_function": resourceCosmosDbSQLUserDefinedFunction(),
```

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by ms-henglu** [comment]
@katbyte hi, I have updated this PR, could you please take another look? Thanks

File: azurerm/internal/services/cosmos/cosmosdb_sql_function_resource.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+	name := d.Get("name").(string)
+	containerId, _ := parse.SqlContainerID(d.Get("container_id").(string))
+	resourceGroup := containerId.ResourceGroup
+	accountName := containerId.DatabaseAccountName
+	containerName := containerId.ContainerName
+	databaseName := containerId.SqlDatabaseName
+	body := d.Get("body").(string)
+
+	id := parse.NewSqlFunctionID(subscriptionId, resourceGroup, accountName, databaseName, containerName, name)
```
Comment (katbyte) [inline azurerm/internal/services/cosmos/cosmosdb_sql_function_resource.go:75]: we don't need to assign the ID parts to variables here?

**Review by katbyte — CHANGES_REQUESTED** [review]
Thanks @ms-henglu - one comment to address and this hsould be good to merge

**Comment by ms-henglu** [comment]
@katbyte , thanks for code review, I updated pr according to your suggestion.

**Review by katbyte — APPROVED** [review]
Thanks @ms-henglu - LGTM 👍

**Pull request merged**